test/test.c

```diff
@@ -3001,16 +3004,37 @@ static hb_dict_t * PreparePreset(const char *preset_name)
             return NULL;
         }
     }
-    else
+    else if (preset_import_gui == 1)
     {
         preset = hb_presets_get_default();
+        if (preset == NULL)
+        {
+            fprintf(stderr, "Error loading presets! Aborting.\n");
+            return NULL;
+        }
+        preset = hb_value_dup(preset);
     }
-    if (preset == NULL)
+    else
     {
-        fprintf(stderr, "Error loading presets! Aborting.\n");
-        return NULL;
+        // Default settings
```

Comment:
One benefit of the explicit "auto" preset is that we presumably wouldn't have to check for null when testing unset values, like in 9d2f3f5.

That said, we should at minimum make it an external JSON and C files pair much like the built-in presets.